src/js/angular/aclmanagement/controllers.js

```diff
@@ -212,6 +251,20 @@ function AclManagementCtrl($scope, $location, toastr, AclManagementRestService,
             });
     };
 
+
+    /**
+     * Function to switch between tabs in the user interface.
+     *
+     * @param {string} scope
+     */
+    $scope.switchTab = (scope) => {
+        if ($scope.editedRuleIndex !== undefined) {
+            $scope.cancelEditing($scope.activeTabScope, $scope.editedRuleIndex)
+        }
+        $scope.activeTabScope = scope;
+        $scope.ruleKeys = Object.keys($scope.rulesModel.filterByScope($scope.activeTabScope)[0] || {});
```

Comment:
Can you move the obtaining of the keys for the rule of given type in a method in the model too?

Reply:
done